### PR TITLE
Fix (hopefully) uniqueness for gateway AcrPull role assignment

### PR DESCRIPTION
**Pull request by jewzaam**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10743419

### What this PR does / why we need it:

For some reason one region deployment failed for non-unique GUID in AcrPull role assignment.  In investigating I don't see a great reason why it should fail.  But GUIDs are not guarenteed unique.  This PR simply cleans up the GUID generation as we need to change _something_ for this region.

### Test plan for issue:

E2E, verify in INT, verify in Canary.

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

**Comment by mjudeikis** [comment]
GUID in ARM is not unique. Its a static hash based on inputs.
> The returned value isn't a random string, but rather the result of a hash function on the parameters. The returned value is 36 characters long. It isn't globally unique. To create a new GUID that isn't based on that hash value of the parameters, use the newGuid function.
https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#guid

This change does not look right. First it changes existing naming patterns (without cleaning old RBAC), 

And this:

"name": "[concat(substring(parameters('acrResourceId'), add(lastIndexOf(parameters('acrResourceId'), '/'), 1)), '/', '/Microsoft.Authorization/', guid(concat(parameters('acrResourceId'), **parameters('gatewayServicePrincipalId')**, 'RP / AcrPull')))]"
vs
 "name": "[guid(concat(parameters('acrResourceId'), **parameters('gatewayServicePrincipalId')**, 'Gateway / AcrPull'))]",
looks like we are removing data, but I dont see how this will make it more unique based on guid behavior. 


Second, Error I think is race condition for multiple parallel runs so conflict is coming from there (I think) 
we have this:
https://github.com/Azure/ARO-RP/blob/18439dff653218914ab74811c164063cc6dff757/pkg/deploy/predeploy.go#L243-L266

Which deploys this: 
https://github.com/Azure/ARO-RP/blob/18439dff653218914ab74811c164063cc6dff757/pkg/deploy/generator/resources_rp.go#L1641

Can it be similar failure mode where same resource is being modified in parallel?

**Comment by jewzaam** [comment]
@mjudeikis agree on GUID uniqueness.  Don't agree on removing data.  It stops the parsing of the acrReousrceId to build some mocked up and wrong scope for the guid generation.  It's just wrong, not useful, and complex.  If we want to use scoping (which I don't know how it can be anything other than hashing the input, it is not efficient to check true uniqueness) we might pass subscription as the first argument to `guid`

On parallel execution we have a single retry.  It's being used.  It is only this one region that is having a problem.  Given that guid generation is not unique I am suspecting we are hitting a case where generation is creating a duplicate.  If we suspect parallel deployments can we stop them from being parallel?  What does it gain us?

On cleanup, yes understood.  This isn't worth the effort to automate cleanup in my opinion.  We have a small number of regions to manually cleanup.  We're already using elevated access to create certs.  This cleanup can be done while waiting for those scripts to run for the next sector.  Or when we run for japaneast as due to the issue this is trying to fix the gateway keyvault is not yet created and we'll need to create certs.

File: deploy/rp-production-global.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
             ]
         },
         {
-            "name": "[concat(substring(parameters('acrResourceId'), add(lastIndexOf(parameters('acrResourceId'), '/'), 1)), '/', '/Microsoft.Authorization/', guid(concat(parameters('acrResourceId'), parameters('gatewayServicePrincipalId'), 'RP / AcrPull')))]",
```
Comment (mjudeikis) [inline deploy/rp-production-global.json:73]: `"name": "[concat(substring(parameters('acrResourceId'), add(lastIndexOf(parameters('acrResourceId'), '/'), 1)), '/', '/Microsoft.Authorization/', guid(concat(parameters('acrResourceId'), parameters('gatewayServicePrincipalId'), 'Gateway / AcrPull')))]"`

**Review by mjudeikis — APPROVED** [review]

**Review by nilsanderselde — APPROVED** [review]

**Pull request merged**